WcaOnRails/app/models/competition.rb

```diff
@@ -1512,4 +1512,11 @@ def to_ics
     cal.publish
     cal
   end
+
+  validate :advancement_condition_must_be_present_for_all_non_final_rounds, if: :confirmed_or_visible?
+  def advancement_condition_must_be_present_for_all_non_final_rounds
+    if !rounds.all?(&:advancement_condition_is_valid?)
```

Comment:
Nit: this is a perfect use case for `unless rounds.all...` ;)
Also if `rounds` had a significant size, doing `if rounds.any?(!cond)` is slightly better than `if !rounds.all?(cond)`, because  `any?` stops at the first match, whereas `all?` has to iterate through everything.